src/isomorphic/classic/element/__tests__/ReactElementValidator-test.js

```diff
@@ -286,35 +286,48 @@ describe('ReactElementValidator', () => {
     );
   });
 
-  it('gives a helpful error when passing null, undefined, boolean, or number', () => {
+  it('gives a helpful error when passing invalid types', () => {
     spyOn(console, 'error');
     React.createElement(undefined);
     React.createElement(null);
     React.createElement(true);
     React.createElement(123);
-    expectDev(console.error.calls.count()).toBe(4);
+    React.createElement({x: 17});
+    React.createElement({});
+    expectDev(console.error.calls.count()).toBe(6);
     expectDev(console.error.calls.argsFor(0)[0]).toBe(
-      'Warning: React.createElement: type should not be null, undefined, ' +
-      'boolean, or number. It should be a string (for DOM elements) or a ' +
-      'ReactClass (for composite components).'
+      'Warning: React.createElement: type is invalid -- expected a string ' +
+      '(for built-in components) or a class/function (for composite ' +
+      'components) but got: undefined.'
```

Comment:
Maybe it would be nice to also add the export warning for `undefined`, since trying to import a named export that doesn't exist will return `undefined` and I think that's probably a common mistake too. Especially when using third party libraries.